setup.py

```diff
@@ -2,34 +2,34 @@
 import itertools
 from setuptools import setup, find_packages
 
-__version__ = runpy.run_path('elf/__version__.py')['__version__']
+__version__ = runpy.run_path("elf/__version__.py")["__version__"]
 
 requires = [
     "numpy",
     "imageio",
     "scikit-image",
-    "skan"
+    "skan",
+    "h5py",
```

Comment:
h5py and zarr fortunately exist on both pip and conda, but are they hard requirements? It might be nice to keep them optional as this library will be useful to people doing lots of different things and they might not want those behemoths lying around (and taking time out of their builds).

I like the idea of having "functional" extras, used in particular workflows, and then additionally have the conda/pip extras for a maximal install on either of those platforms.

Reply:
> It might be nice to keep them optional as this library will be useful to people doing lots of different things and they might not want those behemoths lying around (and taking time out of their builds).

Good point.

> I like the idea of having "functional" extras, used in particular workflows, and then additionally have the conda/pip extras for a maximal install on either of those platforms.

I am not using pip so much, so I haven't used the extra feature much. But sure can't hurt to support additional dependencies.
We can do something like this:
```python
requires = ["numpy", "imageio", "scikit-image", "skan"]
# functional extras for pip
extras = {"hdf5": "h5py",
              "zarr": "zarr",
              "n5": "pyn5"}
# dependencies available on conda only
conda_only = ["z5py", "nifty", "vigra"]
# collect all dependencies for conda (don't support pyn5 and don't need zarr)
conda_exclude = ["zarr", "pyn5"]
conda_all = conda_only + [v for v in extras.values() if v not in conda_exclude] 
extras["conda_all"] = conda_all

setup(..., extra_requirements=extras)
```

This way, all extra functional dependencies would be available in pip, except for `conda_all`.
In case we want to support different flavors in conda later, this could be done via more `conda_` keys.

Reply:
Looks great! Should be able to drop the extra travis lines and update the `meta.yaml`, check that builds, merge this PR into the other PR, and we're away.